### PR TITLE
Main page articles 

### PR DESCRIPTION
**Pull request by tyndria**
screencapture-localhost-3000-2018-03-13-13_33_22

Add Footer, Header, SpecialHeading, LanguageSwitcher (mostly without redux logic)
Add components for structuring articles previews on main page

Add Diary component (it's on progress in branch 'diary')

### CODE REVIEW

File: utils/customPropTypes.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import PropTypes from 'prop-types';
+
+const brandPropTypes = {
```
Comment (Drapegnik) [inline utils/customPropTypes.js]: imho better: `export const BrandModel = `

File: utils/customPropTypes.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  slug: PropTypes.string.isRequired,
+  name: PropTypes.string.isRequired,
+};
+
+const articlePropTypes = {
```
Comment (Drapegnik) [inline utils/customPropTypes.js]: `export const ArticleModel`

File: components/articles/ArticlePreview.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import articlePropTypes from 'utils/customPropTypes';
```
Comment (Drapegnik) [inline components/articles/ArticlePreview.js]: `import { ArticleModel } from `

File: components/articles/Diary.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    <article className="card tile is-child notification">
+      <h1 className="title">Дзённік дня</h1>
+      <div className="date">
+        <span>{date}</span>
+      </div>
+      <span
+        className="icon is-small arrow arrow-left"
+        role="button"
+        tabIndex="0"
+        onKeyDown={e => e.keyCode === LEFT_ARROW_KEY && getPrevDiary()}
```
Comment (Drapegnik) [inline components/articles/Diary.js:18]: For example more general approach (can be extracted to parent component):
```js
const actions = {
	[LEFT_ARROW_KEY]: getPrevDiary,
	[RIGHT_ARROW_KEY]: getNextDiary,
}

const handleKeyDown = e => {
	const action = actions[e.keyCode];
	if (action) {
		action();
	}
}

...

onKeyDown={handleKeyDown}
```

File: components/articles/grid/ArticlesComplexRow.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import articlePropTypes from 'utils/customPropTypes';
+import ArticleRow from './ArticlesRow';
+
+const ArticlesComplexRow = ({ articles }) => (
+  <div className="second-line tile is-ancestor">
+    <div className="tile is-vertical is-8">
+      <ArticleRow articles={articles.slice(0, 2)} />
```
Comment (Drapegnik) [inline components/articles/grid/ArticlesComplexRow.js]: `0, 2` - please extract to some constant with self-explanatory name

File: components/articles/ArticlePreview.js (hunk 3 above)
Comment (tyndria) [reply components/articles/ArticlePreview.js]: Ok, I will use postfix Model, thanks

File: components/common/Footer.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    link: 'https://instagram.com/wir_by',
+    id: '6',
+  },
+];
+
+const Footer = () => (
+  <footer className="app-footer">
+    <div className="columns">
+      <div className="column general-info">
+        {text.projectGeneralInfo} <i className="fa fa-copyright" /> 2018
```
Comment (Drapegnik) [inline components/common/Footer.js]: i think we can get year with `moment` or just `Date`

File: components/common/Footer.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        <br />
+        {text.readAboutUs}
+        <Link href="/about">
+          <a> {text.here} </a>
+        </Link>
+      </div>
+      <div className="column social">
+        <div>{text.followSocialLinks}</div>
+        <div className="links">
+          {socialNets.length &&
```
Comment (Drapegnik) [inline components/common/Footer.js]: redundant

File: components/common/Header.js [hunk 8]
```diff
@@ -2,24 +2,72 @@ import React from 'react';
...
+            </span>
+          ))}
+      </div>
+
+      <div className="navbar-end">
+        <div className="navbar-item">
+          <div className="user">
+            <span>{user && user.email}</span>
+            <div>
+              <Button onClick={e => signOut(e)} className="button logout is-text">
```
Comment (Drapegnik) [inline components/common/Header.js]: Why `signOut(e)`?

File: components/common/LanguageSwitcher.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const LanguageSwitcher = ({ currentLang, langs, onClick }) => (
+  <div className="select current-lang">
+    <select defaultValue={currentLang.id || langs[0].id}>
+      {langs &&
+        langs.length &&
```
Comment (Drapegnik) [inline components/common/LanguageSwitcher.js]: redundant

File: pages/index.js [hunk 10]
```diff
@@ -34,25 +37,15 @@ class HomePage extends Component {
...
-                      <br />
-                    </div>
-                  ))}
-                <br />
-              </li>
-            ))}
-        </ol>
+        <div className="main-page page-container">
+          <div className="page-content">
+            <ArticlesRow articles={articles.slice(0, 4)} />
```
Comment (Drapegnik) [inline pages/index.js]: same as above, or we can add helper func `getArticleRow(articles)`, for your choice

File: redux/ducks/articles.js [hunk 11]
```diff
@@ -46,9 +46,19 @@ export const actions = {
   }),
 };
 
+const getLocalizedBrand = (brand, lang) => ({ slug: brand.slug, name: brand.names[lang] });
+
+const getLocalizedArticles = (articles, lang = 'be') =>
```
Comment (Drapegnik) [inline redux/ducks/articles.js:51]: 👍 i'm thinking about that method, too

**Review by Drapegnik — CHANGES_REQUESTED** [review]

File: components/articles/Diary.js (hunk 4 above)
Comment (tyndria) [reply components/articles/Diary.js:18]: @Drapegnik , that's good idea, but I don't know what to do with onClick event handling. I could, for example, pass some argument to identify what button was clicked, but I'm not sure that it's really worth doing

**Comment by Drapegnik** [comment]
@tyndria can we merge it?

**Comment by tyndria** [comment]
@Drapegnik , yes, let's merge

**Review by Drapegnik — APPROVED** [review]

**Comment by Drapegnik** [comment]
[main page frontend](https://trello.com/c/ck2rjG3z/51-main-page-frontend)

**Pull request merged**